src/include/liburing.h

```diff
@@ -919,17 +919,23 @@ IOURINGINLINE void io_uring_prep_statx(struct io_uring_sqe *sqe, int dfd,
 	sqe->statx_flags = (__u32) flags;
 }
 
+static inline __u32 __io_uring_cap_len(size_t len)
+{
+	size_t mask = -(size_t)(len > UINT32_MAX);
+	return (len & ~mask) | (UINT32_MAX & mask);
```

Comment:
Based on https://godbolt.org/z/zGeqGGMeh, I'd suggest

```suggestion
	return len > UINT32_MAX ? UINT32_MAX : len;
```

Reply:
I had this back and forth with @axboe already. I think the cmov or branching version is better than the current version too (and moreover it'll leverage max instructions for architectures that have them, like aarch64). But since @axboe has stated that he prefers the bit twiddling version I'll let him decide.

Reply:
That's not how I understood https://github.com/axboe/liburing/pull/1436#issuecomment-3067197709 , given that it's replying to a comment that explicitly mentions cmov. But I'll let him comment.